### PR TITLE
Poll for inventory in v2 add_device_to_inventory

### PR DESCRIPTION
**Pull request by grybak-arista**
It can take a moment for device to appear. This polls the inventory for 60 seconds to see if the device is added.

### CODE REVIEW

File: cvprac/cvp_api.py [hunk 1]
```diff
@@ -361,6 +362,48 @@ def get_inventory(self, start=0, end=0, query=''):
...
+
+            if wait:
+                # It can take a few moments for devices to appear in inventory
+                dev = None
+                timeout = time.time() + 60      # Try for 60 seconds
+                while not dev and time.time() < timeout:
+                    devices = self.get_inventory()
+                    for device in devices:
+                        if ('ipAddress' in device and
+                                device['ipAddress'] == device_ip):
```
Comment (mharista) [inline cvprac/cvp_api.py]: Does this work? Where is device_ip defined?

File: cvprac/cvp_api.py [hunk 2]
```diff
@@ -361,6 +362,48 @@ def get_inventory(self, start=0, end=0, query=''):
...
+                            dev = device
+                            break
+                    if not dev:
+                        time.sleep(5)
+
+                if not dev:
+                    # Device did not show up in inventory
+                    raise RuntimeError('Device {} failed to appear in '
+                                       'undefined inventory'
+                                       .format(device_ip))
```
Comment (mharista) [inline cvprac/cvp_api.py]: Where is device_ip defined?

File: cvprac/cvp_api.py [hunk 3]
```diff
@@ -361,6 +362,48 @@ def get_inventory(self, start=0, end=0, query=''):
...
+                'add_devices_to_undefined: not implemented for v1 APIs')
+        else:
+            self.log.debug('v2 Inventory API Call')
+            data = {'hosts': device_ips}
+            self.clnt.post('/inventory/devices', data=data,
+                           timeout=self.request_timeout)
+
+            if wait:
+                # It can take a few moments for devices to appear in inventory
+                dev = None
```
Comment (mharista) [inline cvprac/cvp_api.py]: This function take multiple device addresses to be added to undefined but only checks that one of them has been added during the wait?

**Comment by devrobo** [comment]
Are we sure we want the library to poll?

Gary,
Do you need this in the test environment because the topology just spun up and you are waiting for the devices to appear? This shouldn't be needed during normal operation correct?

**Comment by grybak-arista** [comment]
Updated changes and submitted as a new pull request.

**Pull request closed without merging**